### PR TITLE
Change to looped transactions

### PR DESCRIPTION
**Pull request by s3364859**
@6nop @s3364942 @s3494110 @s3482247 

In this PR:
 - End-to-end test of all XML data sources
 - Change transactions to looped transactions of 100k records per transaction

clapping-gif

### CODE REVIEW

File: CPT331.Data.Parsers/Parser.cs [hunk 1]
```diff
@@ -22,19 +23,32 @@ public Parser(string fileName)
...
+				List<Crime> crimesToCommit = crimes.Take(toTake).ToList();
+
+				StringBuilder stringBuilder = new StringBuilder();
+
+				stringBuilder.AppendLine();
+				stringBuilder.AppendLine("BEGIN TRAN");
+				stringBuilder.AppendLine();
 
-			crimes.ForEach(m => stringBuilder.AppendLine($"EXEC Crime.spAddCrime @LocalGovernmentAreaID = {m.LocalGovernmentAreaID}, @OffenceID = {m.OffenceID}, @Count = {m.Count}, @Month = {m.Month}, @Year = {m.Year}"));
+				crimesToCommit.ForEach(m => stringBuilder.AppendLine($"EXEC Crime.spAddCrime @LocalGovernmentAreaID = {m.LocalGovernmentAreaID}, @OffenceID = {m.OffenceID}, @Count = {m.Count}, @Month = {m.Month}, @Year = {m.Year}"));
```
Comment (s3364859) [inline CPT331.Data.Parsers/Parser.cs]: Some of this code keeps delivery nuggets.

File: CPT331.Data.Migration/CPT331.Data.Migration.csproj [hunk 2]
```diff
@@ -23,6 +23,7 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <Prefer32Bit>true</Prefer32Bit>
```
Comment (s3364859) [inline CPT331.Data.Migration/CPT331.Data.Migration.csproj]: Not sure why this is added - don't really care though tbh.

Comment (6nop) [inline CPT331.Data.Migration/CPT331.Data.Migration.csproj]: Yeah, that's weird. You're not using a budgo x86 PC are you? haha 😁

File: CPT331.Data.Parsers/NswXmlParser.cs [hunk 3]
```diff
@@ -37,19 +37,19 @@ protected override void OnParse(string fileName, List<Crime> crimes)
 				string offenceName = xmlNode.ChildNodes[1].InnerText.Trim();
 				string suboffenceName = xmlNode.ChildNodes[2].InnerText.Trim();
 
-				Console.WriteLine($"{localGovernmentAreaName}: {offenceName} ({suboffenceName})");
+				//	Console.WriteLine($"{localGovernmentAreaName}: {offenceName} ({suboffenceName})");
 
 				Offence offence = null;
-				LocalGovernmentArea localGovernmentArea = localGovernmentAreas.Where(m => (m.Name == localGovernmentAreaName)).FirstOrDefault();
+				LocalGovernmentArea localGovernmentArea = localGovernmentAreas.Where(m => (m.Name.ToUpper() == localGovernmentAreaName.ToUpper())).FirstOrDefault();
```
Comment (6nop) [inline CPT331.Data.Parsers/NswXmlParser.cs]: Would `m.Name.Equals(localGovernmentAreaName, StringComparison.InvariantCultureIgnoreCase)` be nicer?

Comment (6nop) [inline CPT331.Data.Parsers/NswXmlParser.cs]: We could abstract this complication our with an extension method?

```
public class ThingyMajiggy  
{  
   public static EqualsIgnoreCase(this string compareFrom, string compareTo)  
   {    
      return compareFrom.Equals(compareTo,  
                         StringComparison.InvariantCultureIgnoreCase);  
   }  

}  
```

File: CPT331.Core/Extensions/StringExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+#endregion
+
+namespace CPT331.Core.Extensions
+{
+	public static class StringExtensions
+	{
+		public static bool EqualsIgnoreCase(this string value, string compareTo)
+		{
+			return value.Equals(compareTo, StringComparison.OrdinalIgnoreCase);
```
Comment (s3364859) [inline CPT331.Core/Extensions/StringExtensions.cs]: So I believe that OrdinalIgnoreCase is fastest for this kind of comparison where language mechanics don't really matter.

File: CPT331.Data.Migration/CPT331.Data.Migration.csproj [hunk 5]
```diff
@@ -23,6 +23,7 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <Prefer32Bit>false</Prefer32Bit>
```
Comment (s3364859) [inline CPT331.Data.Migration/CPT331.Data.Migration.csproj]: Err...

File: CPT331.Data.Parsers/NswXmlParser.cs (hunk 3 above)
Comment (s3364859) [inline CPT331.Data.Parsers/NswXmlParser.cs]: This is a great idea, and is what a non-lazy dev should be doing. Stand by for the next commit...

**Comment by 6nop** [comment]
I'm happy with this PR. Hitting the button.
image button

**Pull request merged**